packages/compass-schema-validation/src/modules/rules-generation.ts

```diff
@@ -119,6 +126,16 @@ export const rulesGenerationReducer: Reducer<RulesGenerationState, Action> = (
     };
   }
 
+  if (
+    isAction(action, ValidationActions.ValidatorChanged) &&
+    state.isGenerated
+  ) {
+    return {
+      ...state,
+      isGenerated: false, // the generated validator has been changed
```

Comment:
Probably fine as is, do we also want to include the `ValidationLevelChanged` and `ValidationActionChanged`, also maybe `ValidationSaveStarted`?